[CLD-6659] Add documentation for Cloud IP Filtering (#6792)
mattermost/docs: source/manage/cloud-ip-filtering.rst

Cloud IP Filtering
========================

.. include:: ../_static/badges/ent-cloud-only.rst
   :start-after: :nosearch:

> IP filtering is a powerful security feature that allows system admins to control access to their workspace by defining approved IP ranges. Only users within these specified IP ranges can access the workspace, ensuring enhanced security for your workspace. IP filtering requires a subscription to Mattermost Cloud Enterprise.  

Reviewer: I removed capitalization on ``filtering`` in ``IP Filtering`` in this PR but if we want the docs to treat this as an official product feature name, happy to bring the capitalization back.

Author: I'm 0/5. Spoke with @spirosoik and he's fine with lowercase as well.

> IP Filtering is a powerful security feature that allows system administrators to control access to their workspace by defining approved IP ranges. Only users within these specified IP ranges can access the workspace, ensuring enhanced security for your workspace. IP Filtering requires a subscription to Mattermost Cloud Enterprise.  

Reviewer: ```suggestion
IP filtering is a powerful security feature that allows system admins to control access to their workspace by defining approved IP ranges. Only users within these specified IP ranges can access the workspace, ensuring enhanced security for your workspace. IP filtering requires a subscription to Mattermost Cloud Enterprise.  
```

> Accessing the IP Filtering Configuration:

Reviewer: ```suggestion
Configure IP filtering
```

------------------------

> 1. **Login as System Administrator:** Access the system console of your workspace, ensuring your user is a system administrator.

Reviewer: ```suggestion
1. **Log in as System Admin**: Access the System Console of your workspace, ensuring your user is a system admin.
```

> 2. **Navigate to Site Configuration:** Once logged in, navigate to the "Site Configuration" section within the admin console.

Reviewer: ```suggestion
2. **Go to Site Configuration**: Once logged in, go to the **Site Configuration** section.
```

> 3. **Access IP Filtering Settings:** Under "Site Configuration," select "IP Filtering" to access the IP Filtering settings.

Reviewer: ```suggestion
3. **Access IP Filtering Settings**: Under **Site Configuration**, select **IP Filtering** to access the IP Filtering settings.
```

About CIDR notation
----------------------------

CIDR (Classless Inter-Domain Routing) notation is used to specify a range of IP addresses. It consists of an IP address followed by a forward slash and a number indicating the network's prefix length. For example:

>    - `192.168.0.0/24` represents the IP range from `192.168.0.0` to `192.168.0.255`.

Reviewer: ```suggestion
   - ``192.168.0.0/24`` represents the IP range from ``192.168.0.0`` to ``192.168.0.255``.
```

>    - The `/24` signifies that the first 24 bits are the network address, leaving 8 bits for host addresses.

Reviewer: ```suggestion
   - The ``/24`` signifies that the first 24 bits are the network address, leaving 8 bits for host addresses.
```

> For a more in-depth explanation of CIDR notation, refer to `this article </https://aws.amazon.com/what-is/cidr/>`__.

Author: This feature is inherently technical in that it requires knowledge of CIDR notation. I tried to hit an even ground of explanation without rewriting wikipedia. Generally, if you don't know what CIDR notation is, you probably aren't the intended audience for this feature

Reviewer: Very fair assumption to make. Mattermost product documentation focuses on Mattermost-centric details and points readers to external, third-party sources where appropriate. Nicely done.

> Configuring IP Filters:

Reviewer: ```suggestion
Configure IP Filters
```

------------------------

> Adding an IP Range:

Reviewer: ```suggestion
Add an IP Range
```

~~~~~~~~~~~~~~~~~~~~~~~
To add an IP range to the whitelist, follow these steps:

1. Select the **Add Filter** button within the IP Filtering settings page in the System Console.

> 2. Enter the IP range using CIDR notation. For example, `192.168.1.0/24`.

Reviewer: ```suggestion
2. Enter the IP range using CIDR notation. For example, ``192.168.1.0/24``.
```

> 1. Click on the "Add Filter" button within the IP Filtering settings page in the System Console.

Reviewer: ```suggestion
1. Select the **Add Filter** button within the IP Filtering settings page in the System Console.
```

3. Provide a descriptive name or label for the IP range to ease identification in the future.
4. Save the changes.

.. note::

   The system console will restrict you from saving changes if the IP address you are accessing your workspace on is not within the ranges you have specified at the time of clicking save

> Editing or Removing an Existing IP Range:

Reviewer: ```suggestion
Edit or remove an existing IP range
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To edit or remove an existing IP range from the whitelist:

> 1. Locate the IP range you wish to modify within the IP Filtering settings.

Reviewer: ```suggestion
1. Locate the IP range you want to modify within the **IP Filtering** settings.
```

> 2. Hover over the rule you'd like to edit or delete, and click on the respective edit or delete option beside the IP range.

Reviewer: ```suggestion
2. Hover over the rule you'd like to edit or delete, and select the respective edit or delete option beside the IP range.
```

3. Make necessary changes or confirm the removal of the IP range.

> 4. Save changes by clicking the "Save" button

Reviewer: ```suggestion
4. Save your changes by selecting **Save**.
```

Enable/disable IP filtering
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
System admins have the option to enable or disable IP filtering:

- **Enable:** To activate IP filtering, ensure at least one IP range is added to the whitelist.
- **Disable:** Temporarily disable IP filtering by removing all IP ranges from the whitelist, or by flipping the global IP Filtering toggle in the System Console.

> Unable to access your workspace

Reviewer 2: 100% we need to be explicit (with console images) here to help in the case that the workspace owner is not technical *AND* they are responding to an issue users are unable to access the cloud workspace.

> - **Disable:** Temporarily disable IP Filtering by removing all IP ranges from the whitelist, or by flipping the global IP Filtering toggle in the System Console.

Reviewer: ```suggestion
- **Disable:** Temporarily disable IP filtering by removing all IP ranges from the whitelist, or by flipping the global IP Filtering toggle in the System Console.
```

> Unable to Access Your Workspace

Reviewer: ```suggestion
Unable to access your workspace
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> If  you are unable to access your workspace due to previously set IP filters, and you need to regain access to your workspace, your workspace owner can:

Reviewer: ```suggestion
If you are unable to access your workspace due to previously set IP filters, and you need to regain access to your workspace, your workspace owner can:
```

> 1. Navigate to the `Mattermost Customer Portal </https://customers.mattermost.com/>`__.

Reviewer: ```suggestion
1. Visit the `Mattermost Customer Portal </https://customers.mattermost.com/>`__.
```

> 2. Click on the "IP Filtering" menu item in the left hand side bar.

Reviewer: ```suggestion
2. Select the **IP Filtering** menu item in the left hand side bar.
```

> 3. Click on the button to "Disable IP Filtering"

Reviewer: ```suggestion
3. Select **Disable IP Filtering**.
```

Cloud IP Filtering
========================

.. include:: ../_static/badges/ent-cloud-only.rst
   :start-after: :nosearch:

IP filtering is a powerful security feature that allows system admins to control access to their workspace by defining approved IP ranges. Only users within these specified IP ranges can access the workspace, ensuring enhanced security for your workspace. IP filtering requires a subscription to Mattermost Cloud Enterprise.

> Configure IP filtering

Reviewer: @nickmisasi - Based on the [v9.4 server changelog](https://github.com/mattermost/docs/pull/6825), Cloud IP Filtering will be disabled OOTB behind a feature flag. 

Are you open to updating this new docs page to include details on enabling the feature flag as an admin?

Author: End users can't control feature flags in Cloud. The feature flag's in place so that we can wait until other system dependencies (CWS, Provisioner, etc) are ready without having to hold up the 9.4 release. An additional benefit is that we can disable the new feature in the event a glaring issue is found after release. 

I'm not sure it's worth documenting this as it's invisible to the user and may just cause confusion. Let me know what you think!

Author: One thing to add - the plan would be to enable the feature flag immediately as soon as the release rolls out to the enterprise ring in Cloud

Reviewer: You're absolutely right -- users can't enable feature flags in Cloud. I'd forgotten that important point. No sense documenting something that Cloud admins won't be able to do.

Given that the plan is to enable soon after release to the Enterprise ring, I wonder if the changelog needs a small update to clarify that point? Thoughts?
